test/numa/run.sh

```diff
@@ -0,0 +1,499 @@
+    for py_assertion in "$@"; do
+        speed=1000 out "### Verifying assertion '$py_assertion'"
+        ( speed=1000 host-command "python3 -c '${py_cpus}; assert($py_assertion)'" >/dev/null 2>&1 && out "### The assertion holds." ) || {
+                out "### The assertion FAILED in context ${py_cpus}"
+                echo "verify: assertion '$py_assertion' failed when ${py_cpus}" >> "$SUMMARY_FILE"
+                echo "    verify command output: $COMMAND_OUTPUT"
+                exit 1
+        }
+    done
+}
+
+delete() { # script API
+    # Usage: delete PARAMETERS
+    #
+    # Run "kubectl delete PARAMETERS".
+    vm-command "kubectl delete $*" || {
+        command-error "kubectl delete failed"
+    }
+}
+
+create() { # script API
+    # Usage: [VAR=VALUE] create TEMPLATE_NAME
+    #
+    # Create n instances from TEMPLATE_NAME.yaml.in, copy each of them
+    # from host to vm, kubectl create -f them, and wait for them
+    # becoming Ready.
+    #
+    # Parameters:
+    #   TEMPLATE_NAME: the name of the template without extension (.yaml.in)
+    #
+    # Optional parameters (VAR=VALUE):
+    #   wait: condition to be waited for (see kubectl wait --for=condition=).
+    #         If empty (""), skip waiting. The default is wait="Ready".
+    #   wait_t: wait timeout. The default is wait_t=60s.
+    template_name=$1.yaml.in
+    local template_kind
+    template_kind=$(awk '/kind/{print tolower($2)}' < "$template_name")
+    local wait=${wait-Ready}
+    local wait_t=${wait_t-60s}
+    if [ -z "$n" ]; then
+        local n=1
+    fi
+    if [ ! -f "$template_name" ]; then
+        error "error creating \"$1\": missing template ${template_name}"
+    fi
+    for _ in $(seq 1 $n); do
+        kind_count[$template_kind]=$(( ${kind_count[$template_kind]} + 1 ))
+        local NAME="${template_kind}$(( ${kind_count[$template_kind]} - 1 ))" # the first pod is pod0
+        eval "echo -e \"$(<"${template_name}")\"" > "$NAME.yaml"
+        host-command "scp $NAME.yaml $VM_SSH_USER@$VM_IP:" || {
+            command-error "copying $NAME.yaml to VM failed"
+        }
+        vm-command "cat $NAME.yaml"
+        vm-command "kubectl create -f $NAME.yaml" || {
+            command-error "kubectl create error"
+        }
+        if [ "x$wait" != "x" ]; then
+            speed=1000 vm-command "kubectl wait --timeout=${wait_t} --for=condition=${wait} ${template_kind}/$NAME" >/dev/null 2>&1 || {
+                command-error "waiting for ${template_kind} \"$NAME\" to become ready timed out"
+            }
+        fi
+    done
+}
+
+interactive() { # script API
+    # Usage: interactive
+    #
+    # Enter the interactive mode: read next script commands from
+    # the standard input until "exit".
+    echo "Entering the interactive mode until \"exit\"."
+    while true; do
+        # shellcheck disable=SC2162
+        ( while read -e -p "run.sh> " -a commands; do
+              eval "${commands[@]}"
+          done
+        ) && break
```

Comment:
I tried to get `rlwrap` working so that commands could have been executed in the current shell process. Couldn't get it working. Tried something like `rlwrap -a -H /tmp/delme.history bash -c 'while read -r -e -a cmd; do echo "$cmd[@]" >> /tmp/delme.fifo; done' &`... so I'm pushing without rlwrap.

Reply:
You have omit the `-e` option from `read` (to disable the usage of readline), otherwise `rlwrap` will not work. But even if you do that `rlwrap` works somewhat crappily for wrapping bash, so maybe we should just forget about it.